File: internal/domain/customer.go
package domain

import "time"

// Customer represents a business or individual that a sales rep visits.
type Customer struct {
	ID      string       `json:"id"`
	Name    string       `json:"name"`
	Type    CustomerType `json:"type"`
	Address Address      `json:"address"`
	// Phone is the primary contact phone number.
	Phone string `json:"phone"`
	// Email is the primary contact email address.
	Email string `json:"email"`
	// Notes holds free-form notes about the customer.
	Notes     string    `json:"notes"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

// Address holds the physical location of a customer.
type Address struct {
	Street  string `json:"street"`
	City    string `json:"city"`
	State   string `json:"state"`
	Country string `json:"country"`
	Zip     string `json:"zip"`
}

File: web/src/types/customer.ts
/**
 * Customer domain types — mirror the Go backend domain model.
 * Field names use camelCase (frontend convention), mapping to the
 * snake_case json tags on the Go domain.Customer struct.
 */

export type { CustomerType } from './lead'

import type { CustomerType } from './lead'

export interface Address {
  street: string
  city: string
  state: string
  country: string
  zip: string
}

export interface Customer {
  id: string
  name: string
  type: CustomerType
  address: Address
  phone: string
  email: string
  notes: string
  createdAt: string
  updatedAt: string
}

export interface CreateCustomerInput {
  name: string
  type: CustomerType
  address?: Address
  phone?: string
  email?: string
  notes?: string
}

export interface UpdateCustomerInput {
  id: string
  name?: string
  type?: CustomerType
  address?: Address
  phone?: string
  email?: string
  notes?: string
}

export interface CustomerListParams {
  page?: number
  limit?: number
  type?: CustomerType
}
